doc/devel/coding_guide.rst

```diff
@@ -1,369 +1,312 @@
-.. _pr-guidelines:
+.. _coding_guidelines:
+
+*****************
+Coding guidelines
+*****************
 
-***********************
-Pull request guidelines
-***********************
+While the current state of the Matplotlib code base is not compliant with all
+of these guidelines, our goal in enforcing these constraints on new
+contributions is that it improves the readability and consistency of the code base
+going forward.
```

Comment:
Nowadays we are mostly in line with our guidelines, so IMHO this can be removed.